src/quest_manager/models.py

```diff
@@ -19,12 +19,47 @@
 from tags.models import TagsModelMixin
 
 
+class CategoryManager(models.Manager):
+    """
+    Custom manager for Category model that provides
+    queryset methods for active campaigns with available quests.
+    """
+
+    def all_active_with_importable_quests(self):
+        """
+        Returns a queryset of active campaigns (categories) that have
+        at least one quest visible to students and not archived.
+
+        This helps efficiently filter campaigns relevant for display
+        or import in the library context.
+        """
+
+        # Start with all categories filtered to only those marked active
+        qs = self.get_queryset()
+
+        # Annotate each category with a count of related quests that:
+        # - are visible to students (not drafts)
+        # - are not archived (still active)
+        qs = qs.annotate(
+            current_quest_count=Count(
+                'quest',
+                filter=Q(quest__visible_to_students=True, quest__archived=False)
+            )
+        )
+
+        # Filter out categories that do not have any qualifying quests
+        qs = qs.filter(current_quest_count__gt=0)
+
+        return qs
+
+
 class Category(IsAPrereqMixin, models.Model):
     """ Used to group quests into 'Campaigns'
     """
     title = models.CharField(max_length=50, unique=True)
     icon = models.ImageField(upload_to='icons/', null=True, blank=True)
     short_description = models.CharField(max_length=500, blank=True, null=True)
+    objects = CategoryManager()
```

Comment:
`objects`  should go after all the fields, with a line break. Check out the other models for consistancy